### PR TITLE
mention isAutomatedPublishing

### LINKED ISSUE #17553: Add documentation for `isAutomatedPublishing` option for `java-gradle-plugin`
Currently, No documentation for `isAutomatedPublishing` option for `java-gradle-plugin` in [the userguide for `java-gradle-plugin`][userguide_java_gradle_plugin], It's hard to know How to disable automatically generated plugin marker artifact (the publications ends with PluginMarkerMaven) and plugin artifact (the publication named pluginMaven). I think `isAutomatedPublishing` should be documented in the user guide.

### Expected Behavior

There's the documentation for `isAutomatedPublishing` in the user guide.

### Current Behavior

No documentations for `isAutomatedPublishing` in the user guide.

[userguide_java_gradle_plugin]: https://docs.gradle.org/current/userguide/java_gradle_plugin.html

### PR DESCRIPTION
**Pull request by anatawa12**
Fixes #17553 

### Context

Some users want to publish some Gradle plugins to maven central but the auto-generated plugin marker artifact doesn't follow [Maven Central Requirements][central-requirements] so we want to disable auto-generation of plugin marker artifact. However, we can't know how to disable it from the user guide. This enables us to know how to disable it from the user guide.

### Contributor Checklist
- [x] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md)
- [x] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [x] Make sure all contributed code can be distributed under the terms of the [Apache License 2.0](https://github.com/gradle/gradle/blob/master/LICENSE), e.g. the code was written by yourself or the original code is licensed under [a license compatible to Apache License 2.0](https://apache.org/legal/resolved.html).
- [x] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team
- [ ] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective
- [ ] Provide unit tests (under `<subproject>/src/test`) to verify logic
- [ ] Update User Guide, DSL Reference, and Javadoc for public-facing changes
- [ ] Ensure that tests pass sanity check: `./gradlew sanityCheck`
- [x] Ensure that tests pass locally: `./gradlew <changed-subproject>:quickTest`

### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation
- [ ] Recognize contributor in release notes

[central-requirements]: https://central.sonatype.org/publish/requirements/

### CODE REVIEW

**Comment by big-guy** [comment]
@bot-gradle help

**Comment by tresat** [comment]
@bot-gradle test this

**Review by big-guy — APPROVED** [review]

File: subprojects/docs/src/docs/release/notes.md [hunk 1]
```diff
@@ -7,6 +7,7 @@ We would like to thank the following community members for their contributions t
 Include only their name, impactful features should be called out separately below.
  [Some person](https://github.com/some-person)
 -->
+[Shou Kakerigawa(翳河翔)](https://github.com/anatawa12)
```
Comment (anatawa12) [inline subprojects/docs/src/docs/release/notes.md:10]: Can you add as anatawa12? Shou Kakerigawa is also a handle name but not for programming.

**Comment by tresat** [comment]
@bot-gradle test and merge

File: subprojects/docs/src/docs/release/notes.md (hunk 1 above)
Comment (tresat) [reply subprojects/docs/src/docs/release/notes.md]: Of course!  I'll merge that change for you.

**Comment by big-guy** [comment]
@bot-gradle test and merge

**Pull request merged**